http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -0,0 +1,316 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package service
+
+import java.time.Instant
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.effect.concurrent.Semaphore
+import cats.effect.{Async, Blocker, ContextShift, Timer}
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import com.google.auth.oauth2.AccessToken
+import com.google.cloud.BaseServiceException
+import io.chrisdavenport.log4cats.StructuredLogger
+import org.broadinstitute.dsde.workbench.google2.util.RetryPredicates
+import org.broadinstitute.dsde.workbench.google2.{GcsBlobName, GoogleStorageService}
+import org.broadinstitute.dsde.workbench.leonardo.RuntimeImageType.{Jupyter, Welder}
+import org.broadinstitute.dsde.workbench.leonardo.config.{AutoFreezeConfig, DataprocConfig, GceConfig, ImageConfig}
+import org.broadinstitute.dsde.workbench.leonardo.dao.DockerDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, DbReference, SaveCluster}
+import org.broadinstitute.dsde.workbench.leonardo.http.api.{CreateRuntime2Request, RuntimeServiceContext}
+import org.broadinstitute.dsde.workbench.leonardo.http.service.LeonardoService._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.RuntimeServiceInterp._
+import org.broadinstitute.dsde.workbench.leonardo.model._
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.CreateCluster
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{google, TraceId, UserInfo, WorkbenchEmail}
+
+import scala.concurrent.ExecutionContext
+
+class RuntimeServiceInterp[F[_]](blocker: Blocker,
+                                 semaphore: Semaphore[F],
+                                 config: RuntimeServiceConfig,
+                                 authProvider: LeoAuthProvider[F],
+                                 serviceAccountProvider: ServiceAccountProvider[F],
+                                 dockerDAO: DockerDAO[F],
+                                 googleStorageService: GoogleStorageService[F],
+                                 publisherQueue: fs2.concurrent.Queue[F, LeoPubsubMessage])(implicit F: Async[F],
```

Comment:
Just curious why have 'explicit' implicits here rather than `F[_]: Async: Logger: Timer: ContextShift` (not sure I have opinion, just trying to learn the convention).

Reply:
I found a little annoying to always type `Async[F].delay` or `Logger[F].info` etc...I think it makes more sense to have them as context bounds if the function body itself is not directly referencing them...There's a library that allows as to do things like `F.delay` when using context bound, but I haven't looked at it yet (will paste a link if I find it again...saw it on twitter)....